dwl.c

```diff
@@ -1742,9 +1744,23 @@ rendermon(struct wl_listener *listener, void *data)
 	struct timespec now;
 	clock_gettime(CLOCK_MONOTONIC, &now);
 
-	/* Do not render if any XDG clients have an outstanding resize. */
+
+			/* Do not render if any XDG clients have an outstanding resize. */
 	wl_list_for_each(c, &stack, slink) {
-		if (c->resize) {
+		if (m->resize) {
+			if (c->resize) {
+				wlr_surface_send_frame_done(client_surface(c), &now);
+				render = 0;
+			}
+		}
+
+		if (c->surface.xdg->role != WLR_XDG_SURFACE_ROLE_TOPLEVEL)
```

Comment:
Here you need to check if the client is xdg or xwayland, `createnotify()` does this check, so it's not possible to have any xdg client other than `WLR_XDG_SURFACE_ROLE_TOPLEVEL`

Reply:
This part is actually still from #20 and was used to prevent a segfault. We don't need this here

Reply:
Although you are right. We have to check if it's an xdg or xwayland surface

Reply:
I just noticed that dwl can't seems to handle xwayland resizes at all. It abosutely sucks (i mean main branch)